### ISSUE TITLE
Training on TPU is very slow

### ISSUE THREAD

**Issue by StrangerZhang** [michuanhaohao/reid-strong-baseline#144]
Training on TPU is very slow due to the naive triplet loss hard samples mining

**Comment by michuanhaohao** [comment]
We did not test our code on TPU. The project is based on the standard Pytorch. It is sorry that we cannot give you any help. Maybe the sampler of triplet loss is slow on TPU.

**Comment by StrangerZhang** [comment]
Fix it! The bottleneck of the training code is the addmm_ which is slow on TPU, and the other time consuming op is advanced index in triplet hard mining code. I will make an pr later, so this nice work can train on google cloud TPU

**Comment by michuanhaohao** [comment]
OK, thank you for your contributions. You can create a new branch or add ‘TPU’ options into the config file. Please be careful not to affect the training on the GPU.
